core/types/calls.cc

```diff
@@ -1363,6 +1363,26 @@ class Magic_buildArray : public IntrinsicMethod {
     }
 } Magic_buildArray;
 
+class Magic_buildRange : public IntrinsicMethod {
+public:
+    void apply(const GlobalState &gs, DispatchArgs args, const Type *thisType, DispatchResult &res) const override {
+        if (args.args.empty()) {
+            res.returnType = Types::rangeOfUntyped();
+            return;
+        }
+
+        auto type_member = Types::dropLiteral(args.args[0]->type);
+
+        if (type_member->isNilClass()) {
+            type_member = Types::untypedUntracked();
```

Comment:
I'm not sure this case is necessary -- why not just use the if below that checks if it's a subtype of `NilClass` instead?

Reply:
Or, why not always dropNil if it's a subtype of `NilClass`, and then return untypedUntracked if it collapses to bottom?

Reply:
Good idea 👍